crates/aptos-log-derive/src/lib.rs

```diff
@@ -147,6 +147,7 @@ fn extract_internal_type(ty: &Type) -> Option<&Type> {
     None
 }
 
+#[allow(clippy::never_loop)]
```

Comment:
What is this one?

Reply:
There's a weird loop pattern in this function (it's unclear to me what the author was trying to do). I've updated it according to the clippy (https://rust-lang.github.io/rust-clippy/master/index.html#/never_loop). I'm trying to minimize any potential breakages 😄